utils/kubernetes/kube_state_processor.py

```diff
@@ -152,63 +239,34 @@ def kube_deployment_spec_replicas(self, message, **kwargs):
+    # Labels attached: deployment, namespace
+    def kube_deployment_spec_paused(self, message, **kwargs):
+        """ Indicates if the deployment is paused. """
+        metric_name = NAMESPACE + '.deployment.paused'
         for metric in message.metric:
             val = metric.gauge.value
             tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
-            #TODO: add deployment/replicaset?
             self.gauge(metric_name, val, tags)
 
-    # Labels attached: namespace, pod, container, node
-    def kube_pod_container_requested_memory_bytes(self, message, **kwargs):
-        """ Memory bytes requested for a container in a pod. """
-        metric_name = NAMESPACE + '.container.memory_requested'
+    # Labels attached: deployment, namespace
+    def kube_deployment_spec_strategy_rollingupdate_max_unavailable(self, message, **kwargs):
+        """ Maximum number of pods unavailable during rolling updates. """
+        metric_name = NAMESPACE + '.deployment.rollingupdate_max_unavailable'
         for metric in message.metric:
             val = metric.gauge.value
             tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
             self.gauge(metric_name, val, tags)
 
-    # TODO: Uncomment after kube-state-metrics 0.4 is released
-    # Labels attached: namespace, pod, container, node
-    # def kube_pod_container_limits_cpu_cores(self, message, **kwargs):
-    #     """ CPU cores limit for a container in a pod. """
-    #     metric_name = NAMESPACE + '.container.cpu_limit'
-    #     for metric in message.metric:
-    #         val = metric.gauge.value
-    #         tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
-    #         #TODO: add deployment/replicaset?
-    #         self.gauge(metric_name, val, tags)
-
-    # TODO: Uncomment after kube-state-metrics 0.4 is released
-    # Labels attached: namespace, pod, container, node
-    # def kube_pod_container_limits_memory_bytes(self, message, **kwargs):
-    #     """ Memory byte limit for a container in a pod. """
-    #     metric_name = NAMESPACE + '.container.memory_limit'
-    #     for metric in message.metric:
-    #         val = metric.gauge.value
-    #         tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
-    #         self.gauge(metric_name, val, tags)
-
-    # TODO: implement kube_pod_container_status_ready
-    # TODO: implement kube_pod_container_status_running
-    # TODO: implement kube_pod_container_status_terminated
-    # TODO: implement kube_pod_container_status_waiting
-
-    # Labels attached: namespace, pod, container
-    def kube_pod_container_status_restarts(self, message, **kwargs):
-        """ Number of desired pods for a deployment. """
-        metric_name = NAMESPACE + '.container.restarts'
+    # Labels attached: deployment, namespace
+    def kube_deployment_metadata_generation(self, message, **kwargs):
```

Comment:
same as the other `metadata_generation` one, do you have a use case where it could be useful? Otherwise let's not collect it.